rust/fastsim-core/src/traits.rs

```diff
@@ -19,9 +19,14 @@ pub trait SerdeAPI: Serialize + for<'a> Deserialize<'a> {
     /// # Returns:
     ///
     /// A Rust Result
-    fn to_file(&self, filename: &str) -> Result<(), anyhow::Error> {
+    fn to_file(&self, filename: &str) -> anyhow::Result<()> {
         let file = PathBuf::from(filename);
-        match file.extension().unwrap().to_str().unwrap() {
+        match file
+            .extension()
+            .ok_or_else(|| anyhow!("Unable to parse file extension: {:?}", file))?
```

Comment:
I think you could change 
```rust
ok_or_else(|| anyhow!("Unable to parse file extension: {:?}", file))?
```
to something like
```rust
ok_or(|e| anyhow!("{e}\nUnable to parse file extension: {:?}", e, file))?
```
to preserve whatever the native error message is as context.

Reply:
Hm, I don't know that the native error message really exists. The `unwrap` was on an `Option`, not a `Result`, so the error was just about unwrapping on a `None`